src/chunk_insert_state.c

```diff
@@ -385,7 +388,7 @@ setup_on_conflict_state(ChunkInsertState *state, ChunkDispatch *dispatch,
 
 		if (NULL == chunk_map)
 			chunk_map = convert_tuples_by_name(RelationGetDescr(chunk_rel),
-											   RelationGetDescr(first_rel)
+											   RelationGetDescr(hyper_rel)
```

Comment:
The fix is not clear to me: we create a chunk map here if one is not available.  Wouldn't we still run into this if we have more than 2 columns (the test case has only 1 additional column) : i.e. we have a chunk map but some of the columns are missing in the mapping?

Reply:
the fix is on line 156 this is just a cosmetic change

Reply:
the chunk_map passed as argument will only be non-NULL if we have RETURNING clause otherwise it will always be NULL cause caller didn't bother building the map

Reply:
the code would certainly benefit from some more refactoring/cleanup but i wanted to get the bugfix out

Reply:
ok, I see that the map is null if no conversion is required. Yes, confusing to see calls to translate when it is not applicable.